fmriprep/config.py

```diff
@@ -471,18 +471,23 @@ def init(cls):
             _db_path.mkdir(exist_ok=True, parents=True)
 
             # Recommended after PyBIDS 12.1
+            ignore_patterns = [
+                'code',
+                'stimuli',
+                'sourcedata',
+                'models',
+                re.compile(r'^\.'),
+                re.compile(r'sub-[a-zA-Z0-9]+(/ses-[a-zA-Z0-9]+)?/(beh|dwi|eeg|ieeg|meg|perf)'),
+            ]
+            if cls.participant_label:
+                # Ignore any subjects who aren't the requested ones.
+                ignore_patterns.append(
+                    re.compile(r'sub-(?!' + '|'.join(cls.participant_label) + r')\w+')
+                )
```

Comment:
I think you probably don't want to do this if `cls.bids_database_dir is not None`. This would cause a database file to be written that is missing large portions of the dataset and then reused the next time.

Reply:
I thought the BIDSLayout's `reset_database` setting meant that the database would be left untouched if `bids_database_dir` was set?

Reply:
Otherwise, wouldn't it be a problem if users wanted to run both fMRIPrep (which ignores `dwi` folders) and dMRIPrep/QSIPrep (which needs them) on the same dataset?

Reply:
The issue is that if `--bids-database-dir` is set to a path where there does not already exist a file, it will be created and then frozen with the first round.